### PR TITLE
feat(looker): represent looker explores as assets

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
Model the basic Looker explore where an explore is joined off of existing Looker views.

To parse the Looker specifications, we use [`lkml`](https://lkml.readthedocs.io/en/latest/simple.html).

A Looker explore will always be based on an existing sql table. By default, this is the `explore_name`, but it can be overridden with the [`from`](https://cloud.google.com/looker/docs/reference/param-explore-from) attribute.

Joins on this table are specified using [`join`](https://cloud.google.com/looker/docs/reference/param-explore-join).

The combination of these two lists will give the upstream dependencies for the Looker view.

## How I Tested These Changes
pytest

### CODE REVIEW

File: python_modules/libraries/dagster-looker/dagster_looker/asset_decorator.py [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
 from pathlib import Path
-from typing import Any, Callable
+from typing import Any, Callable, Mapping, Sequence
 
+import lkml
 import yaml
 from dagster import AssetKey, AssetsDefinition, AssetSpec, multi_asset
 
 
+def build_looker_explore_specs(project_dir: Path) -> Sequence[AssetSpec]:
```
Comment (benpankow) [inline python_modules/libraries/dagster-looker/dagster_looker/asset_decorator.py:10]: do we plan on eventually having some sort of translator API to customize the asset key specs built?

**Review by benpankow — APPROVED** [review]
LGTM

File: python_modules/libraries/dagster-looker/dagster_looker/asset_decorator.py (hunk 1 above)
Comment (rexledesma) [reply python_modules/libraries/dagster-looker/dagster_looker/asset_decorator.py:10]: Yeah. That will be a separate PR

**Comment by rexledesma** [comment]
### Merge activity

* **May 7, 11:20 AM PDT**: [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/21635) rebased this pull request as part of a merge.

**Pull request merged**